Plugin/Quote/Model/BillingAddressManagement.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Yotpo\SmsBump\Plugin\Quote\Model;
+
+use Magento\Quote\Model\BillingAddressManagement as QuoteBillingAddressManagement;
+use \Magento\Quote\Api\Data\AddressInterface;
+
+/**
+ * Class BillingAddressManagement
+ * Plugin to trigger checkout sync
+ */
+class BillingAddressManagement extends AbstractCheckoutTrigger
+{
+    /**
+     * @method afterAssign
+     * @param  QuoteBillingAddressManagement $billingAddressManagement
+     * @param  int                           $result
+     * @param  int                           $cartId
+     * @param  AddressInterface              $address
+     * @param  boolean                       $useForShipping
+     * @return int
+     */
+    public function afterAssign(QuoteBillingAddressManagement $billingAddressManagement, $result, $cartId, AddressInterface $address, $useForShipping = false)
+    {
+        if (
+            !$this->registry->registry('yotpo_smsbump_quote_set_billing_address_plugin') &&
+            $this->yotpoMessagingConfig->isCheckoutSyncActive()
+        ) {
+            $quote = $this->quoteRepository->getActive($cartId);
+            $this->checkoutSync($quote);
+            $this->registry->unregister('yotpo_smsbump_quote_set_billing_address_plugin');
+        }
```

Comment:
are we sure this is correct? it checks if `registry->registry('yotpo_smsbump_quote_set_billing_address_plugin')` is null or empty, and if so it unregister it aftewards? `registry->unregister('yotpo_smsbump_quote_set_billing_address_plugin')`

Looking at the implementation it returns 'mixed' type, is this method returns boolean?
![image](https://user-images.githubusercontent.com/10895971/161089941-f6388008-2036-4671-849d-9f7402e928db.png)

Reply:
Yeah, it's a mistake, the unregister() should have been outside the if.
Pushed a fix for that to the same branch (https://github.com/YotpoLtd/magento2-module-messaging/commit/dc3b606ea53f3c12d1a6beebd163907f0ee179c8)
*Also, it's worth mentioning that this is a workaround that most systems won't use.